.github/workflows/build.yml

```diff
@@ -0,0 +1,127 @@
+            carthage_platform: macos
+          - scheme: "PactConsumerSwift tvOS"
+            destination: "OS=14.2,name=Apple TV 4K (at 1080p)"
+            carthage_platform: tvos
+
+    env:
+      SCHEME: ${{ matrix.scheme }}
+      DESTINATION: ${{ matrix.destination }}
+      CARTHAGE_PLATFORM: ${{ matrix.carthage_platform }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Use Xcode 12.2
+        run: sudo xcode-select -switch /Applications/Xcode_12.2.app
+
+      - name: "Cache dependencies"
+        uses: actions/cache@v2
+        with:
+          path: Carthage/build
+          key: ${{ runner.os }}-pact-${{ hashFiles('**/Cartfile.resolved') }}
+          restore-keys: |
+            ${{ runner.os }}-pact-
+            ${{ runner.os }}-
+
+      - name: Prepare the tools
+        run: |
+          scripts/install_deps.sh
+
+      - name: "Run tests"
+        run: |
+          scripts/build.sh
+
+      - name: "Upload code coverage"
+        run: |
+          bash <(curl -s https://codecov.io/bash) -J 'PactConsumerSwift'
+
+  test_spm:
+    name: Test for SPM compatibility
+    runs-on: macOS-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Use Xcode 12.2
+        run: sudo xcode-select -switch /Applications/Xcode_12.2.app
+
+      - name: Swift build
+        run: |
+          swift build
+          echo "⚠️  Skipping \"swift test\" as no test target is defined in \"Package.swift\" (https://github.com/DiUS/pact-consumer-swift/commit/229f35d63a547f492c7ba9e177ac8d7b685e7a7f)"
+
+  test_carthage:
+    needs: [test_xcodebuild]
+    name: "Test Carthage dependency"
+    runs-on: macOS-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Use Xcode 12.2
+        run: sudo xcode-select -switch /Applications/Xcode_12.2.app
+
+      - name: "Cache dependencies"
+        uses: actions/cache@v2
+        with:
+          path: Carthage/build
+          key: ${{ runner.os }}-pact-${{ hashFiles('**/Cartfile.resolved') }}
+          restore-keys: |
+            ${{ runner.os }}-pact-
+            ${{ runner.os }}-
+
+      - name: Prepare Tools
+        run: |
+          scripts/install_deps.sh
+
+      - name: Carthage build
+        run: |
+          scripts/carthage_xcode12 build --no-skip-current --platform "ios,macos,tvos"
+
+  after_success:
+    needs: [test_xcodebuild, test_spm, test_carthage]
+    name: "Build example projects"
+    runs-on: ubuntu-20.04
+
+    env:
+      AUTH_TOKEN: ${{ secrets.AUTH_TOKEN }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Trigger Demo Project builds
+        run: |
+          scripts/build_children.sh
```

Comment:
This script does not return non-zero if it fails. Instead of a script file we could have a strategy matrix that lists the Demo projects' urls to trigger once they are moved away from TravisCI.org. Similar to [this](https://github.com/surpher/PactSwift/blob/main/.github/workflows/continuous_integration.yml#L140).